grid2op/tests/test_ma.py

```diff
@@ -0,0 +1,518 @@
+                                                                                     False, False, False, False, False, 
+                                                                                     True, True, True,False, False])).all()
+        
+        assert (self.ma_env._action_domains['agent_0']['mask_line_or'] == self.ma_env._action_domains['agent_0']['mask_line_ex']).all()
+        assert (self.ma_env._action_domains['agent_1']['mask_line_or'] == self.ma_env._action_domains['agent_1']['mask_line_ex']).all()
+        
+        assert (self.ma_env._action_domains['agent_0']['mask_shunt'] == [False]).all()
+        assert (self.ma_env._action_domains['agent_1']['mask_shunt'] == [True]).all()
+        
+    def test_interco(self):
+        
+        self.ma_env = MultiAgentEnv(self.env, self.action_domains, _add_to_name="test_interco")
+        # Tests on interconnections with known values
+        assert (self.ma_env._action_domains['agent_0']['mask_interco'] == [False, False, False, False, False, False, False, False, False,
+                                                                            False, False, False, False, False, False,  True,  True,  True,
+                                                                            False, False]).all()
+        assert (self.ma_env._action_domains['agent_1']['mask_interco'] == [False, False, False, False, False, False, False, False, False,
+                                                                            False, False, False, False, False, False,  True,  True,  True,
+                                                                            False, False]).all()
+        assert (self.ma_env._action_domains['agent_0']['interco_is_origin'] == [True, True, True]).all()
+        assert (self.ma_env._action_domains['agent_1']['interco_is_origin'] == np.invert([True, True, True])).all()
+        
+        # In the two-agent case, they must have the same number of interconnections
+        assert self.ma_env._subgrids_cls['action']['agent_0'].n_interco == self.ma_env._subgrids_cls['action']['agent_1'].n_interco
+        
+        # In the two-agent case, the total number of lines in the global env 
+        # must be equal to the number of line in both local grids plus
+        # the number of interconnections.
+        assert self.ma_env._subgrids_cls['action']['agent_0'].n_line \
+             + self.ma_env._subgrids_cls['action']['agent_1'].n_line \
+             + self.ma_env._subgrids_cls['action']['agent_0'].n_interco == self.env.n_line
+                
+
+    def test_build_subgrid_obj(self):
+        """test the MultiAgentEnv._build_subgrid_obj_from_domain"""
+        
+        # 1
+        action_domains = {
+            'agent_0' : [0,1,2,3, 4],
+            'agent_1' : [5,6,7,8,9,10,11,12,13]
+        }
+        # run redispatch agent on one scenario for 100 timesteps
```

Comment:
???

Why redispatch ? No you can't do this at all you should not be able to take any action at the moment

Reply:
Deleted it